docs/reference/settings/ml-settings.asciidoc

```diff
@@ -80,6 +80,11 @@ total memory of the machine, not current free memory. Jobs are not allocated to
 a node if doing so would cause the estimated memory use of {ml} jobs to exceed
 the limit.
 
+`xpack.ml.max_ml_node_size`::
+(<<static-cluster-setting,Static>>)
+The maximum node size for {ml} nodes. Defaults to `0b`, which means this value
+is ignored.
+
```

Comment:
This is an advanced setting that should only be set if the way that ES is deployed allows for automatic cluster scaling. This setting should be set to the maximum possible future node. This way when assigning a ML job to a node (and it is lazily assignable), we can fail quickly as we know that scaling will never support a job of this size.

Reply:
Thanks, I've taken another crack at adding those details.